src/GafferImage/ImageWriter.cpp

```diff
@@ -1282,6 +1294,74 @@ ImageSpec createImageSpec( const ImageWriter *node, const ImageOutput *out, cons
 	return spec;
 }
 
+// Remove leading, trailing, or repeated dot seperators
+std::string cleanExcessDots( std::string name )
+{
+	size_t repeatedPos = name.find( ".." );
+	while( repeatedPos != std::string::npos )
+	{
+		name.erase( repeatedPos, 1 );
+		repeatedPos = name.find( ".." );
+	}
+
+	if( name[0] == '.' )
+	{
+		name.erase( 0, 1 );
+	}
+
+	if( name[name.size()-1] == '.' )
+	{
+		name.erase( name.size()-1, 1 );
+	}
+	
+	return name;
+}
+
+// Get the EXR names for the view, part, and channel for a Gaffer channel
+std::tuple< std::string, std::string, std::string > viewPartChannelName( const ImageWriter *node, const std::string &view, const std::string &layerAndChannel )
+{
+	std::string layer = "";
+	std::string channel = "";
+	size_t p = layerAndChannel.find( "." );
+	if( p != string::npos )
+	{
+		layer = layerAndChannel.substr( 0, p );
+		channel = layerAndChannel.substr( p + 1 );
+	}
+	else
+	{
+		channel = layerAndChannel;
+	}
+
+	// "naming:hogwashChannel" is an exr channel name matching Nuke - in complete contradiction to the spec,
+	// layers other than the main one have the channel names RGBA written as "red", "green", "blue", "alpha"
+	string hogwashChannel = channel;
+	if( layer.size() )
+	{
+		if( hogwashChannel == "R" ) hogwashChannel = "red";
+		else if( hogwashChannel == "G" ) hogwashChannel = "green";
+		else if( hogwashChannel == "B" ) hogwashChannel = "blue";
+		else if( hogwashChannel == "A" ) hogwashChannel = "alpha";
+	}
+
+	// For use in naming parts, it's useful to have a layer name that is set to "rgba" for channels that
+	// are in the default layer.
+	std::string layerWithDefault = layer.size() ? layer : "rgba";
+
+	Context::EditableScope namingContext( Context::current() );
+	namingContext.set( "naming:view", &view );
+	namingContext.set( "naming:layer", &layer );
+	namingContext.set( "naming:channel", &channel );
+	namingContext.set( "naming:hogwashChannel", &hogwashChannel );
+	namingContext.set( "naming:layerWithDefault", &layerWithDefault );
+
+	return std::make_tuple( 
+		cleanExcessDots( node->namingConventionViewNamePlug()->getValue() ),
+		cleanExcessDots( node->namingConventionPartNamePlug()->getValue() ),
+		cleanExcessDots( node->namingConventionChannelNamePlug()->getValue() )
+	);
```

Comment:
While `hogwashChannel` makes me chuckle every time I see it, I think we need to come up with something that communicates our opinion a little more gently :)

Here's a few thoughts on the naming generally :

- It's a bit misleading that `channel` means "the bit after the last dot" here when everywhere else in Gaffer it means "the whole channel name". I propose using the same nomenclature as `ImageAlgo` and calling it `baseName`.
- The `naming:` prefix seems pretty vague in the wider context of Gaffer, when most nodes prefix with the node name or something close (`loop:`, `collect:` etc). A couple of possible alternatives :
    - `imageWriter:channelName`, `imageWriter:baseName` etc
    - `image:channelName` (we already have this as standard), `image:channelName:baseName`, `image:channelName:layerName`. Perhaps we already ruled out reusing `image:channelName` though?
- `layerWithDefault` isn't that self-explanatory and is only used as a part name in the presets, so perhaps `defaultPartName` would be better?
- I propose `imageWriter:nukeBaseName` as a replacement for `hogwashChannel`